prov/util/src/util_shm.c

```diff
@@ -145,6 +145,35 @@ size_t smr_calculate_size_offsets(size_t tx_count, size_t rx_count,
 	return total_size;
 }
 
+static int smr_retry_map(const char *name, int *fd)
+{
+	char tmp[NAME_MAX];
+	struct smr_region *old_shm;
+	struct stat sts;
+	int ret = FI_SUCCESS;
+
+	*fd = shm_open(name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+	if (*fd < 0)
+		return -errno;
+
+	old_shm = mmap(NULL, sizeof(*old_shm), PROT_READ | PROT_WRITE,
+		       MAP_SHARED, *fd, 0);
+	if (old_shm == MAP_FAILED)
+		return -errno;
+
+	if (!old_shm->pid) {
+		munmap(old_shm, sizeof(*old_shm));
+		return FI_SUCCESS;
+	}
+
+	memset(tmp, 0, sizeof(tmp));
+	snprintf(tmp, sizeof(tmp), "/proc/%d", old_shm->pid);
+	ret = !(stat(tmp, &sts) == -1 && errno == ENOENT);
```

Comment:
This results in ret either being 0 on success or 1 on failure.  In other flows, the function returns -errno on failure.  For consistency, set ret to either 0 on success or some negative error value on failure.